sr_port/zlput_rname.c

```diff
@@ -140,8 +143,9 @@ boolean_t on_stack(rhdtyp *rtnhdr, boolean_t *need_duplicate)
 
 	if (NULL != need_duplicate)
 		*need_duplicate = FALSE;
-	for (fp = frame_pointer; NULL != fp; fp = SKIP_BASE_FRAME(fp->old_frame_pointer))
+	for (fp = frame_pointer; NULL != fp; fp = fp->old_frame_pointer)
 	{
+		fp = SKIP_BASE_FRAME(fp);
```

Comment:
This is another place where one of those bugs likely exists (not tested yet but it does need one as I suspect it is an exploder.

This routine is called when we are replacing a routine and need to know if it is active on the stack somewhere. Previously our search stopped at a callin base frame so if a routine was in use prior to the callin (and on the stack prior to the callin) this routine won't find and signal it is ok to replace the routine when it is not. Flagged as needing a test